### PR TITLE
Add oak_client Rust template

### PR DESCRIPTION
**Pull request by ipetr0v**
This PR adds a template for the Rust client, which will be built upon as part of the https://github.com/project-oak/oak/issues/3643

### CODE REVIEW

**Review by rbehjati — APPROVED** [review]
LGTM. Thanks.

File: oak_client/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+pub struct OakClientBuilder {
+    transport: Box<dyn AsyncTransport>,
+    evidence_provider: Box<dyn EvidenceProvider>,
+    reference_value: ReferenceValue,
+    verifier: Box<dyn Verifier>,
+    crypto_provider: CryptoProvider,
+}
+
+impl OakClientBuilder {
```
Comment (tiziano88) [inline oak_client/src/lib.rs:40]: not sure I understand the point of the builder struct, if it's just created and then immediately used to build a cient. Isn't it simpler to just have a single function?

File: oak_client/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+impl CryptoProvider {
+    fn get_encryptor(&self, _enclave_public_key: &[u8]) -> anyhow::Result<Encryptor> {
+        Ok(Encryptor {})
+    }
+}
+
+struct Encryptor {}
+
+impl Encryptor {
+    fn encrypt(&mut self, _message: &[u8]) -> anyhow::Result<(Vec<u8>, Decryptor)> {
```
Comment (tiziano88) [inline oak_client/src/lib.rs:116]: Could you clarify what this returns?

File: oak_client/src/verifier.rs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+pub struct AmdSevSnpVerifier {}
+
+impl Verifier for AmdSevSnpVerifier {
+    fn verify(
+        &self,
+        _evidence: &Evidence,
+        _reference_value: &ReferenceValue,
+    ) -> anyhow::Result<()> {
+        Ok(())
```
Comment (tiziano88) [inline oak_client/src/verifier.rs:49]: I think there should be a TODO here too.

**Review by tiziano88 — APPROVED** [review]

File: oak_client/src/lib.rs (hunk 2 above)
Comment (ipetr0v) [reply oak_client/src/lib.rs:96]: Done

File: oak_client/src/verifier.rs (hunk 3 above)
Comment (ipetr0v) [reply oak_client/src/verifier.rs:50]: Done

File: oak_client/src/lib.rs (hunk 1 above)
Comment (ipetr0v) [reply oak_client/src/lib.rs]: Done
Moved it to `OakClient::create(...)`

**Pull request merged**